frontend/src/pages/AccountManagement/MyGroups.js

```diff
@@ -7,69 +7,116 @@ import { Link, useHistory } from 'react-router-dom';
+      acc.push(existingRecipient); // Add the new recipient to the accumulator.
+    }
+    // Add the grant to the recipient.
+    existingRecipient.options.push({
+      value: recipient.grantId,
+      label: `${recipient.name} - ${recipient.grantNumber}${recipient.programTypes.length > 0 ? ` - ${recipient.programTypes.join(', ')}` : ''}`,
+    });
+    return acc;
+  }, [])
+);
 
 export const GROUP_FIELD_NAMES = {
   NAME: 'new-group-name',
   RECIPIENTS: 'select-recipients-new-group',
   IS_PRIVATE: 'is-private',
+  CO_OWNERS: 'co-owners',
+  INDIVIDUALS: 'individuals',
+  SHARE_WITH_EVERYONE: 'share-with-everyone',
 };
 
 export default function MyGroups({ match }) {
-  const { groupId } = match.params;
-  const [recipients, setRecipients] = useState([]);
-  const [error, setError] = useState(null);
-  const history = useHistory();
-  const [recipientsFetched, setRecipientsFetched] = useState(false);
-
-  // see the comment above "onSubmit" for, well, context
-  const { myGroups, setMyGroups } = useContext(MyGroupsContext);
-
-  const { isAppLoading, setIsAppLoading } = useContext(AppLoadingContext);
-
   const {
     control,
     handleSubmit,
     setValue,
     setError: setFormError,
     errors: formErrors,
+    watch,
   } = useForm({
     defaultValues: {
       [GROUP_FIELD_NAMES.NAME]: '',
       [GROUP_FIELD_NAMES.RECIPIENTS]: [],
-      [GROUP_FIELD_NAMES.IS_PRIVATE]: true,
+      [GROUP_FIELD_NAMES.IS_PRIVATE]: false,
+      [GROUP_FIELD_NAMES.CO_OWNERS]: [],
+      [GROUP_FIELD_NAMES.INDIVIDUALS]: [],
+      [GROUP_FIELD_NAMES.SHARE_WITH_EVERYONE]: null,
     },
   });
 
+  const watchIsPrivate = watch(GROUP_FIELD_NAMES.IS_PRIVATE);
+  const watchShareWithEveryone = watch(GROUP_FIELD_NAMES.SHARE_WITH_EVERYONE);
+  const watchCoOwners = watch(GROUP_FIELD_NAMES.CO_OWNERS);
+  const watchIndividuals = watch(GROUP_FIELD_NAMES.INDIVIDUALS);
+
+  const { groupId } = match.params;
+  const [recipientOptions, setRecipientOptions] = useState([]);
+  const [userOptions, setUserOptions] = useState([]);
+  const [error, setError] = useState(null);
+  const history = useHistory();
+  const [recipientsFetched, setRecipientsFetched] = useState(false);
+  const [usersFetched, setUsersFetched] = useState(false);
+  // see the comment above "onSubmit" for, well, context
+  const { myGroups, setMyGroups } = useContext(MyGroupsContext);
+  const { isAppLoading, setIsAppLoading } = useContext(AppLoadingContext);
+
   useEffect(() => {
     async function getGroup() {
       setIsAppLoading(true);
       try {
-        const existingGroupData = await fetchGroup(groupId);
-        if (existingGroupData) {
-          setValue(GROUP_FIELD_NAMES.NAME, existingGroupData.name);
-          setValue(GROUP_FIELD_NAMES.RECIPIENTS, mapGrants(existingGroupData.grants));
-          setValue(GROUP_FIELD_NAMES.IS_PRIVATE, !existingGroupData.isPublic);
+        const fetchedGroup = await fetchGroup(groupId);
+
+        if (fetchedGroup) {
```

Comment:
considering how many fields are being updated, might make sense to use reset instead of setValue (see how TrainingReportForm does it)

https://github.com/HHS/Head-Start-TTADP/blob/33f7d6757ad982d475870a9aa6d0fda5c26069a3/frontend/src/pages/TrainingReportForm/index.js#L41

Reply:
https://www.react-hook-form.com/api/useform/reset/